arkouda/categorical.py

```diff
@@ -647,12 +649,16 @@ def is_registered(self) -> np.bool_:
         Objects registered with the server are immune to deletion until
         they are unregistered.
         """
-        parts_registered:List[np.bool_] = [getattr(self, p).is_registered() for p in Categorical.RegisterablePieces]
+        parts_registered: List[np.bool_] = [p.is_registered() for p in Categorical._get_components(self)]
         if np.any(parts_registered) and not np.all(parts_registered):  # test for error
             raise RegistrationError(f"Not all registerable components of Categorical {self.name} are registered.")
 
         return np.bool_(np.any(parts_registered))
 
+    def _get_components(self) -> List:
+        return [getattr(self, p) for p in Categorical.RegisterablePieces if p in Categorical.RequiredPieces
+                or getattr(self, p) is not None]
```

Comment:
ditto for this. The code is fine and I understand it, but a helper method with a defined interface coupled with pydoc comments would be helpful for a new Arkouda developer